Doc: define PY_SSIZE_T_CLEAN always (#12794)
python/cpython: Doc/extending/extending.rst

   headers on some systems, you *must* include :file:`Python.h` before any standard
   headers are included.

   It is recommended to define ``PY_SSIZE_T_CLEAN`` always before including

>    ``Python.h``.  See :ref:`parsetuple` what this macro is.

Reviewer: Same suggestion as above.

>    ``Python.h``.  See :ref:`parsetuple` what this macro is.

Reviewer: Apologies, I had suggested a change in the second line earlier, but I didn't know how to get it to render correctly in this window to show both lines.

```suggestion
   ``Python.h``.  See :ref:`parsetuple` for a description of this macro.
```
